token.go

```diff
@@ -29,20 +36,103 @@ type Token struct {
 	Valid     bool                   // Is the token valid?  Populated when you Parse/Verify a token
 }
 
-// Create a new Token.  Takes a signing method
-func New(method SigningMethod) *Token {
-	return NewWithClaims(method, MapClaims{})
+func validateCompressionAlgorithm(algorithm string) error {
+	if algorithm != "DEF" {
+		return fmt.Errorf("unsupported compression algorithm: %v", algorithm)
+	}
+
+	return nil
 }
 
-func NewWithClaims(method SigningMethod, claims Claims) *Token {
-	return &Token{
+func compress(data []byte) ([]byte, error) {
+	var b bytes.Buffer
+	gz := gzip.NewWriter(&b)
+
+	if _, err := gz.Write(data); err != nil {
+		return nil, err
+	}
+	if err := gz.Flush(); err != nil {
+		return nil, err
+	}
+	if err := gz.Close(); err != nil {
+		return nil, err
+	}
+
+	return b.Bytes(), nil
+}
+
+func decompress(data []byte) ([]byte, error) {
+	gz, err := gzip.NewReader(bytes.NewReader(data))
+	if err != nil {
+		return nil, err
+	}
+
+	unzipped, err := ioutil.ReadAll(gz)
+	if err != nil {
+		return nil, err
+	}
+
+	return unzipped, nil
+}
+
+// WithClaims specifies the claims to use with the new token
+func WithClaims(claims Claims) TokenOption {
+	return func(t *Token) error {
+		t.Claims = claims
+		return nil
+	}
+}
+
+// WithCompression specifies the compression algorithm to use with the new token
+func WithCompression(algorithm ...string) TokenOption {
```

Comment:
This was made variadic so that the only supported option (`DEF`) could be made the default, but should there be more options in the future, they can be easily supplied instead (e.g., `WithCompression("LZMA")`). I can make this not take an argument at all, and just set deflate if you prefer, so that you can use `jwt.WithCompression` instead of `jwt.WithCompression()`.